addons/mail/static/src/models/thread_viewer/thread_viewer.js

```diff
@@ -167,12 +167,21 @@ function factory(dependencies) {
                 case 'change-of-thread-cache':
                     hint = { type: hintType };
                     break;
+                case 'chat-window-unfolded':
+                    hint = { type: hintType };
+                    break;
                 case 'current-partner-just-posted-message':
                     hint = {
                         messageId: hintData,
                         type: hintType,
                     };
                     break;
+                case 'home-menu-shown':
+                    hint = { type: hintType };
+                    break;
+                case 'home-menu-hidden':
+                    hint = { type: hintType };
+                    break;
```

Comment:
Since this is the current trend, this feels like "useless" lines of code.

Default should be to do this since all of them are doing this, and just the one doing something particular (adding messageId) should have its own condition.

Reply:
Loosing the default error when using a not definer hintType ?

Reply:
It's not an important error to catch, it's just an early catch of developer typos basically + current version makes it really hard for overrides/other modules to add hints too.

Reply:
& regrouping cases like I did ?

Reply:
That still implies unnecessarily having to add new cases there but it's already much better than copy/pasting default behavior. Probably ok in the scope of this PR.

Ideally only `addComponentHint` should be there. Hint format should be changed to have {type: str, data: obj}, and let whoever calls `addComponentHint` to add whatever it wants in those 2 values, instead of actually "merging" data (of which the only current case is `messageId`) into top level of the hint such as done currently in `_makeComponentHint`.